### PR TITLE
CloudMonitoring: Initial GCM Schema

### LINKED ISSUE #62699: CloudMonitoring: Schema phase 1
Complete GCM schema.

Starter PR [here](https://github.com/grafana/grafana/pull/62118).

### PR DESCRIPTION
**Pull request by aangelisc**
Initial GCM schema. Only FE types have been connected. Backend folder has been renamed to match the frontend.

Fixes #62699

### CODE REVIEW

**Review by alyssajoyner — COMMENTED** [review]
This is looking really good 💪 One question I have - it looks like when the dataquery types file was auto generated it was put into the folder `pkg/tsdb/cloud-monitoring` but there is already a folder `pkg/tsdb/cloudmonitoring`. I think we should rename the folders so we don't have a duplicate unless there is a reason that I am unaware of!

**Review by bossinc — APPROVED** [review]

**Review by bossinc — APPROVED** [review]

File: public/app/plugins/datasource/cloud-monitoring/components/AnnotationQueryEditor.tsx [hunk 1]
```diff
@@ -56,7 +57,12 @@ export const AnnotationQueryEditor = (props: Props) => {
 
   // Use a known query type
   useEffect(() => {
-    if (!Object.values(QueryType).includes(query.queryType)) {
+    if (
+      query.queryType !== QueryType.ANNOTATION &&
+      query.queryType !== QueryType.SLO &&
+      query.queryType !== QueryType.TIME_SERIES_LIST &&
+      query.queryType !== QueryType.TIME_SERIES_QUERY
+    ) {
```
Comment (andresmgot) [inline public/app/plugins/datasource/cloud-monitoring/components/AnnotationQueryEditor.tsx:65]: nit:

```suggestion
    if (!query.queryType || !Object.values(QueryType).includes(query.queryType)) {
```

File: public/app/plugins/datasource/cloud-monitoring/components/QueryEditor.tsx [hunk 2]
```diff
@@ -44,7 +45,12 @@ export const QueryEditor = (props: Props) => {
 
   // Use a known query type
   useEffect(() => {
-    if (!Object.values(QueryType).includes(query.queryType)) {
+    if (
```
Comment (andresmgot) [inline public/app/plugins/datasource/cloud-monitoring/components/QueryEditor.tsx:48]: same nit

File: public/app/plugins/datasource/cloud-monitoring/datasource.ts [hunk 3]
```diff
@@ -327,7 +319,7 @@ export default class CloudMonitoringDatasource extends DataSourceWithBackend<
       return !!query.timeSeriesQuery && !!query.timeSeriesQuery.projectName && !!query.timeSeriesQuery.query;
     }
 
-    if ([QueryType.TIME_SERIES_LIST, QueryType.ANNOTATION].includes(query.queryType)) {
+    if ([QueryType.TIME_SERIES_LIST, QueryType.ANNOTATION].includes(query.queryType ?? QueryType.TIME_SERIES_LIST)) {
```
Comment (andresmgot) [inline public/app/plugins/datasource/cloud-monitoring/datasource.ts:322]:
```suggestion
    if (query.queryType && [QueryType.TIME_SERIES_LIST, QueryType.ANNOTATION].includes(query.queryType)) {
```

File: public/app/plugins/datasource/cloud-monitoring/types/types.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import { DataQuery, SelectableValue } from '@grafana/data';
+import { DataSourceOptions, DataSourceSecureJsonData } from '@grafana/google-sdk';
+
+import { MetricKind } from './query';
+
+export interface CloudMonitoringVariableQuery extends DataQuery {
```
Comment (andresmgot) [inline public/app/plugins/datasource/cloud-monitoring/types/types.ts:6]: some of this are also query-related so not sure if they should be in a different file. I would move everything to an index.ts file, with that you can avoid importing `.../types/types`, which is a bit weird.

**Review by andresmgot — APPROVED** [review]
LGTM! Just some minor comments.

File: public/app/plugins/datasource/cloud-monitoring/types/types.ts (hunk 4 above)
Comment (aangelisc) [reply public/app/plugins/datasource/cloud-monitoring/types/types.ts:6]: Agree it's a bit weird, this is a temporary solution until I can refactor out most of these types 😊

**Pull request merged**